test/e2e/node_exporter_test.go

```diff
@@ -184,3 +184,69 @@ nodeExporter:
 		})
 	}
 }
+
+// This test ensures neccessary collectors stay operational after changing generic options in Node Exporter.
+func TestNodeExporterGenericOptions(t *testing.T) {
+	t.Cleanup(func() {
+		f.MustDeleteConfigMap(t, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      clusterMonitorConfigMapName,
+				Namespace: f.Ns,
+			},
+		})
+	})
+
+	collectorsToCheck := []string{
+		"cpu",
+		"diskstats",
+		"filesystem",
+		"hwmon",
+		"loadavg",
+		"meminfo",
+		"netclass",
+		"netdev",
+		"netstat",
+		"os",
+		"stat",
+		"time",
+		"uname",
+		"vmstat",
+	}
+
+	tests := []struct {
+		name        string
+		config      string
+		argsPresent []string
+	}{
+		{
+			name:   "default config",
+			config: "",
+		},
+		{
+			name: "maxprocs = 1",
+			config: `
+nodeExporter:
+  maxProcs: 1`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(st *testing.T) {
+			f.MustCreateOrUpdateConfigMap(t, configMapWithData(t, test.config))
+
+			for _, nameCollector := range collectorsToCheck {
+
+				f.PrometheusK8sClient.WaitForQueryReturn(
+					t, 5*time.Minute, fmt.Sprintf(`min(node_scrape_collector_success{collector="%s"})`, nameCollector),
+					func(v float64) error {
```

Comment:
Hi @raptorsun, a little confused here. Is `maxProcs`under the `collector`? 
I tried value both 0 and 1, but neither can find the metrics
```
% oc -n openshift-monitoring describe pod node-exporter-2zhmh | grep runtime.gomaxprocs
      --runtime.gomaxprocs=1

% oc -n openshift-monitoring exec -c prometheus prometheus-k8s-1  -- curl -k -H "Authorization: Bearer $token" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query --data-urlencode 'query=node_scrape_collector_success{collector="maxProcs"}' | jq
...
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": []
  }
}

Reply:
Hello @Tai-RedHat , thanks for the review. 
The option `maxProcs` is at the same level of `collector`, here is an example
```
 config.yaml: |

    nodeExporter: 
      maxProcs: 1
```
More details at this ticket: 
https://issues.redhat.com/browse/MON-2894

Reply:
To be clear, the `maxProcs` option doesn't enable a new collector so it doesn't show up in the `node_scrape_collector_success` metric.

Reply:
So, is there any other method to check the `maxProcs` value except check in the pod?  
Otherwise LGTM.

Reply:
> So, is there any other method to check the `maxProcs` value except check in the pod?
> Otherwise LGTM.

So, is there any other method to check the maxProcs value except check in the pod?
Otherwise LGTM.

Reply:
> To be clear, the `maxProcs` option doesn't enable a new collector so it doesn't show up in the `node_scrape_collector_success` metric.

So, is there any other method to check the maxProcs value except check in the pod?
Otherwise LGTM.